src/BitNode/BitNodeMultipliers.ts

```diff
@@ -1,168 +1,173 @@
-  PurchasedServerCost: number;
+  PurchasedServerCost = 1;
 
   /** Influence how much it costs to purchase a server */
-  PurchasedServerSoftcap: number;
+  PurchasedServerSoftcap = 1;
 
   /** Influences the maximum number of purchased servers you can have */
-  PurchasedServerLimit: number;
+  PurchasedServerLimit = 1;
 
   /** Influences the maximum allowed RAM for a purchased server */
-  PurchasedServerMaxRam: number;
+  PurchasedServerMaxRam = 1;
 
   /** Influences the minimum favor the player must have with a faction before they can donate to gain rep. */
-  RepToDonateToFaction: number;
+  RepToDonateToFaction = 1;
 
   /** Influences how much money can be stolen from a server when a script performs a hack against it. */
-  ScriptHackMoney: number;
+  ScriptHackMoney = 1;
 
   /**
    * The amount of money actually gained when script hack a server. This is
    * different than the above because you can reduce the amount of money but
    * not gain that same amount.
    */
-  ScriptHackMoneyGain: number;
+  ScriptHackMoneyGain = 1;
 
   /** Influences the growth percentage per cycle against a server. */
-  ServerGrowthRate: number;
+  ServerGrowthRate = 1;
 
   /** Influences the maximum money that a server can grow to. */
-  ServerMaxMoney: number;
+  ServerMaxMoney = 1;
 
   /** Influences the initial money that a server starts with. */
-  ServerStartingMoney: number;
+  ServerStartingMoney = 1;
 
   /** Influences the initial security level (hackDifficulty) of a server. */
-  ServerStartingSecurity: number;
+  ServerStartingSecurity = 1;
 
   /** Influences the weaken amount per invocation against a server. */
-  ServerWeakenRate: number;
+  ServerWeakenRate = 1;
 
   /** Influences how quickly the player's strength level (not exp) scales */
-  StrengthLevelMultiplier: number;
+  StrengthLevelMultiplier = 1;
 
   /** Influences the power of the gift. */
-  StaneksGiftPowerMultiplier: number;
+  StaneksGiftPowerMultiplier = 1;
 
   /** Influences the size of the gift. */
-  StaneksGiftExtraSize: number;
+  StaneksGiftExtraSize = 0;
 
   /** Influences the hacking skill required to backdoor the world daemon. */
-  WorldDaemonDifficulty: number;
+  WorldDaemonDifficulty = 1;
 
   /** Influences profits from corporation dividends and selling shares. */
-  CorporationSoftcap: number;
+  CorporationSoftcap = 1;
 
   /** Influences number of divisions a corporation can have. */
-  CorporationDivisions: number;
+  CorporationDivisions = 1;
 
-  // Index signature
-  [key: string]: number;
+  constructor(a: PartialRecord<keyof BitNodeMultipliers, number> = {}) {
+    for (const [key, value] of getRecordEntries(a)) this[key] = value;
+  }
 }
 
-/** The multipliers that are influenced by current Bitnode progression. */
-export const BitNodeMultipliers = Object.assign({}, defaultMultipliers);
+/** The multipliers currently in effect */
+export let currentNodeMults = new BitNodeMultipliers();
```

Comment:
Although I like this better, I have a slight concern about it. If anything, anywhere in our codebase is keeping a reference to the BitNodeMultipliers, the switch away from const will cause an issue. I don't know of anything that *would*, but...

Reply:
You still can only do a reassignment on a variable in the originating file, so the only place we need to worry about reassignments is in BitNodeMultipliers, all the other files still should essentially view it as a const

Reply:
The difference is that something could have been holding a reference to BitNodeMultipliers, and before it would have still seen changes due to interior mutability.

Since you renamed the variable, we've seen all the callsites so it's probably fine.